### PR TITLE
enhance same attribute name for different tags

### PR DESCRIPTION
**Pull request by nevermore3**
#### What type of PR is this?
feature

#### Which issue(s) this PR fixes:
close #xxx   

#### What this PR does / why we need it?
Add tag information when getting the attributes of the vertex, 

  
#### Checklist：
- [x] Documentation affected （If need to modify document, please label it.)
- [ ] Incompatible （If it is incompatile, please describle it and label it.）
- [ ] Need to cherry pick （If need to cherry pick to some branchs, please label the destination version(s).)
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:
>                                                                 `

### CODE REVIEW

File: tests/tck/features/match/SameTagPropname.feature [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      | [[:like "wang"->"zhang" @0 {likeness: 91}], [:like "zhang"->"wang" @0 {likeness: 98}]] |
+      | [[:like "wang"->"zhang" @0 {likeness: 91}], [:like "zhang"->"sun" @0 {likeness: 78}]]  |
+      | [[:like "wang"->"zhang" @0 {likeness: 91}], [:like "zhang"->"li" @0 {likeness: 89}]]   |
+      | [[:like "wang"->"li" @0 {likeness: 74}], [:like "li"->"sun" @0 {likeness: 99}]]        |
+      | [[:like "wang"->"li" @0 {likeness: 74}], [:like "li"->"qian" @0 {likeness: 89}]]       |
+      | [[:like "wang"->"li" @0 {likeness: 74}]]                                               |
+      | [[:like "wang"->"zhang" @0 {likeness: 91}]]                                            |
+    When executing query:
+      """
+      match p= (v:player)-[e:like]->(d) where v.student.age > 19 return v.student.score, v.student.name, v.player.name
```
Comment (CPWstatic) [inline tests/tck/features/match/SameTagPropname.feature:190]: Do we support `v{.player.name, .student.name}`?

File: tests/tck/features/match/SameTagPropname.feature [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    When executing query:
+      """
+      match (v:student) where v.student.age > 20 return v.student.name, v
+      """
+    Then the result should be, in any order:
+      | v.student.name | v                                                                                                       |
+      | "s_wang"       | ("wang" :player{age: 22, height: 192.0, name: "p_wang"} :student{age: 22, name: "s_wang", score: 82.0}) |
+    When executing query:
+      """
+      match (v:player)-[e:like]->(d) where v.player.age > 19 return e, v.student.name, v.player.name, v.name
```
Comment (yixinglu) [inline tests/tck/features/match/SameTagPropname.feature:113]: I don't know whether we need to keep the compatibility with older version. My suggest is to report the error to user.

Comment (czpmango) [reply tests/tck/features/match/SameTagPropname.feature:113]: Supporting both expressions `v.tag.prop` and `v.prop` at the same time is confusing.

File: tests/tck/features/match/SameTagPropname.feature (hunk 1 above)
Comment (nevermore3) [reply tests/tck/features/match/SameTagPropname.feature:190]: not support right now

File: tests/tck/features/match/SameTagPropname.feature (hunk 2 above)
Comment (yixinglu) [reply tests/tck/features/match/SameTagPropname.feature:113]: so what's the decision?

Comment (nevermore3) [reply tests/tck/features/match/SameTagPropname.feature:113]: not compatible with the previous version,will modify test case

**Review by yixinglu — COMMENTED** [review]
Need to test the previous usage of `var.prop` is illegal.

File: tests/tck/features/match/Base.feature [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Copyright (c) 2020 vesoft inc. All rights reserved.
 #
 # This source code is licensed under Apache 2.0 License.
+@jmq
```
[the remark this answers is not in the document]
Comment (nevermore3) [reply tests/tck/features/match/Base.feature:4]: ok

File: src/graph/optimizer/rule/IndexScanRule.cpp [hunk 4]
```diff
@@ -383,8 +387,14 @@ Status IndexScanRule::analyzeExpression(Expression* expr,
 template <typename E, typename>
 Status IndexScanRule::addFilterItem(RelationalExpression* expr, FilterItems* items) const {
   // TODO (sky) : Check illegal filter. for example : where c1 == 1 and c1 == 2
-  auto relType = std::is_same<E, EdgePropertyExpression>::value ? Expression::Kind::kEdgeProperty
-                                                                : Expression::Kind::kTagProperty;
+  Expression::Kind relType;
+  if (std::is_same<E, EdgePropertyExpression>::value) {
```
Comment (yixinglu) [inline src/graph/optimizer/rule/IndexScanRule.cpp:391]: if constexpr

Comment (nevermore3) [reply src/graph/optimizer/rule/IndexScanRule.cpp:391]: do not understand , could u explain it

File: tests/tck/features/match/SameTagPropname.feature (hunk 1 above)
Comment (CPWstatic) [reply tests/tck/features/match/SameTagPropname.feature:190]:
```
v.{player.name, student.name}
v{.player.name, .student.name}
```

Comment (CPWstatic) [reply tests/tck/features/match/SameTagPropname.feature:190]:
```
[e:like:serve]
```

File: tests/tck/features/match/SeekById.feature [hunk 5]
```diff
@@ -220,51 +243,35 @@ Feature: Match seek by id
...
       """
     Then a ExecutionError should be raised at runtime: Scan vertices or edges need to specify a limit number, or limit number can not push down.
     When executing query:
       """
       MATCH (v)
       WHERE NOT id(v) IN ['James Harden', 'Jonathon Simmons', 'Klay Thompson', 'Dejounte Murray']
-      RETURN v.name AS Name
-      """
-    Then a ExecutionError should be raised at runtime: Scan vertices or edges need to specify a limit number, or limit number can not push down.
-    When executing query:
```
Comment (Shylock-Hg) [inline tests/tck/features/match/SeekById.feature:233]: Why delete this?

File: tests/tck/features/match/SeekById.intVid.feature [hunk 6]
```diff
@@ -220,44 +243,28 @@ Feature: Match seek by id
...
       """
     Then a ExecutionError should be raised at runtime: Scan vertices or edges need to specify a limit number, or limit number can not push down.
     When executing query:
       """
       MATCH (v)
       WHERE NOT id(v) IN [hash('James Harden'), hash('Jonathon Simmons'), hash('Klay Thompson'), hash('Dejounte Murray')]
-      RETURN v.name AS Name
-      """
-    Then a ExecutionError should be raised at runtime: Scan vertices or edges need to specify a limit number, or limit number can not push down.
-    When executing query:
```
Comment (Shylock-Hg) [inline tests/tck/features/match/SeekById.intVid.feature:233]: ditto

File: tests/tck/features/match/SeekById.feature (hunk 5 above)
Comment (nevermore3) [reply tests/tck/features/match/SeekById.feature:233]: can select the index by introducing tag, this case can get the result, already move this case to line :153

**Review by yixinglu — DISMISSED** [review]

File: tests/tck/features/match/SeekById.feature (hunk 5 above)
Comment (Shylock-Hg) [reply tests/tck/features/match/SeekById.feature:233]: Could change to :
```
      MATCH (v)
      WHERE id(v) IN ['James Harden', 'Jonathon Simmons', 'Klay Thompson', 'Dejounte Murray']
      OR v.age == 23
      RETURN v.name AS Name
```
This is aim to test or logical.

**Review by CPWstatic — DISMISSED** [review]
LGTM. But there are a lot conflicts with another pr which implements the openCypher.

File: tests/tck/features/match/SeekById.feature (hunk 5 above)
Comment (nevermore3) [reply tests/tck/features/match/SeekById.feature:233]: ok

**Review by CPWstatic — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**